### ISSUE TITLE
java.lang.NoClassDefFoundError: io/vertx/lang/rx/TypeArg

### ISSUE THREAD

**Issue by vietj** [vert-x3/vertx-rx#176]
When using RX we get this stack trace:

```
java.lang.NoClassDefFoundError: io/vertx/lang/rx/TypeArg
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at io.vertx.reactivex.core.Vertx.<clinit>(Vertx.java:90)
```

This is due to the fact that since now the `vertx-rx-java2-gen` dependency is `provided` the `vertx-rx-gen` dependency is not transitive anymore and thus needs to be added explicitly.

**Comment by timofeevda** [comment]
Great! This fixes the issue introduces in #172. Thanks!

**Comment by vietj** [comment]
yes, I realised how bad the situation was today actually, doing a 3.6.2 release now
